packages/cfd/src/Containers/__tests__/cfd-password-modal.spec.js

```diff
@@ -0,0 +1,513 @@
+import React from 'react';
+import { Router } from 'react-router';
+import { createBrowserHistory } from 'history';
+import { WS, validPassword } from '@deriv/shared';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import CFDPasswordModal from '../cfd-password-modal';
+
+jest.mock('Stores/connect.js', () => ({
+    __esModule: true,
+    default: 'mockedDefaultExport',
+    connect: () => Component => Component,
+}));
+
+jest.mock('@deriv/account', () => ({
+    SentEmailModal: jest.fn(({ should_show_sent_email_modal }) => (
+        <div>{should_show_sent_email_modal && <span>SentEmailModal</span>}</div>
+    )),
+}));
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    getErrorMessages: jest.fn().mockReturnValue({
+        password_warnings: '',
+        password: jest.fn(),
+    }),
+    isDesktop: jest.fn().mockReturnValue(true),
+    WS: {
+        verifyEmail: jest.fn(),
+    },
+    validPassword: jest.fn().mockReturnValue(true),
+}));
+
+describe('ChangePasswordConfirmation', () => {
+    const mockFn = jest.fn();
+    const history = createBrowserHistory();
+    let modal_root_el;
+    const mock_props = {
+        account_title: '',
+        account_type: {},
+        account_status: {},
+        disableCFDPasswordModal: null,
+        email: '',
+        error_message: '',
+        error_type: '',
+        form_error: '',
+        getAccountStatus: null,
+        history: history,
+        is_eu: false,
+        is_fully_authenticated: false,
+        is_cfd_password_modal_enabled: false,
+        is_cfd_success_dialog_enabled: false,
+        is_dxtrade_allowed: false,
+        platform: '',
+        has_cfd_error: false,
+        landing_companies: {},
+        mt5_login_list: [],
+        cfd_new_account: {},
+        setCFDSuccessDialog: null,
+        setMt5Error: null,
+        submitMt5Password: null,
+        submitCFDPassword: null,
+    };
+
+    beforeAll(() => {
+        modal_root_el = document.createElement('div');
+        modal_root_el.setAttribute('id', 'modal_root');
+        document.body.appendChild(modal_root_el);
+    });
+
+    afterAll(() => {
+        document.body.removeChild(modal_root_el);
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should render create Password modal when valid conditions are met', async () => {
+        const props = {
+            ...mock_props,
+            is_cfd_password_modal_enabled: true,
+            is_cfd_success_dialog_enabled: false,
+            account_status: { status: ['mt5_password_not_set', 'dxtrade_password_not_set'] },
+            getAccountStatus: mockFn,
+            platform: 'mt5',
+        };
+        render(
+            <Router history={history}>
+                <CFDPasswordModal {...props} />
+            </Router>
+        );
+        expect(await screen.findByTestId('create-password')).toBeInTheDocument();
+    });
```

Comment:
please check here and further in the tests the correct naming of test ID according to our doc. If this name was created before we should change it to correct one.

Reply:
Will change as per the guide